src/supervisor/watchers/handlers/index.ts

```diff
@@ -91,9 +93,71 @@ const workloadWatchMetadata: Readonly<IWorkloadWatchMetadata> = {
     },
     listFactory: (namespace) => () => k8sApi.appsClient.listNamespacedStatefulSet(namespace),
   },
+  [WorkloadKind.DeploymentConfig]: {
+    /** https://docs.openshift.com/container-platform/4.7/rest_api/workloads_apis/deploymentconfig-apps-openshift-io-v1.html */
+    endpoint: '/apis/apps.openshift.io/v1/watch/namespaces/{namespace}/deploymentconfigs',
+    handlers: {
+      [DELETE]: deploymentConfigWatchHandler,
+    },
+    listFactory: (namespace) => () => k8sApi.customObjectsClient.listNamespacedCustomObject("apps.openshift.io", "v1", namespace, "deploymentconfigs"),
+  },
 };
 
-export function setupInformer(namespace: string, workloadKind: WorkloadKind) {
+async function isSupportedWorkload(
+  namespace: string,
+  workloadKind: WorkloadKind,
+): Promise<boolean> {
+  if (workloadKind !== WorkloadKind.DeploymentConfig) {
+    return true;
+  }
+
+  try {
+    const pretty = undefined;
+    const continueToken = undefined;
+    const fieldSelector = undefined;
+    const labelSelector = undefined;
+    const limit = 1; // Try to grab only a single object
+    const resourceVersion = undefined; // List anything in the cluster
+    const timeoutSeconds = 10; // Don't block the snyk-monitor indefinitely
+    const attemptedApiCall = await k8sApi.customObjectsClient.listNamespacedCustomObject(
+      'apps.openshift.io',
+      'v1',
+      namespace,
+      'deploymentconfigs',
+      pretty,
+      continueToken,
+      fieldSelector,
+      labelSelector,
+      limit,
+      resourceVersion,
+      timeoutSeconds,
+    );
+    return (
+      attemptedApiCall !== undefined &&
+      attemptedApiCall.response !== undefined &&
+      attemptedApiCall.response.statusCode !== undefined &&
+      attemptedApiCall.response.statusCode > 200 &&
+      attemptedApiCall.response.statusCode < 300
+    );
+  } catch (error) {
+    logger.info(
+      { error, workloadKind },
+      'Failed on Kubernetes API call to list DeploymentConfig',
+    );
+    return false;
+  }
+}
+
+export async function setupInformer(namespace: string, workloadKind: WorkloadKind): Promise<void> {
+  const isSupported = await isSupportedWorkload(namespace, workloadKind);
```

Comment:
💡 
But it would complicate the code a bit, we now need to check this only once and we need to differentiate between snyk-monitor scanning the cluster or snyk-monitor scanning a single namespace (won't have permission to query the cluster if scanning a single namespace).

Reply:
Opting not to do it because it would complicate the code quite a bit, LMK if you think it's a really bad idea and the benefits would outweigh the downsides 😉